js-miniapp-sdk/src/miniapp.ts

```diff
@@ -81,9 +81,42 @@ interface Platform {
   getPlatform();
 }
 
+/**
+ * Interfaces to retrieve User profile related information
+ */
+export interface UserInfoProvider {
+  /**
+   * @returns Username saved in the host app user profile
+   */
+  getUserName(): Promise<string>;
+
+  /**
+   * @returns Profile photo saved in the host app user profile
+   */
+  getProfilePhoto(): Promise<string>;
+}
+
+/** @internal */
+class UserInfo implements UserInfoProvider {
+  private bridge: MiniAppBridge;
+
+  constructor(miniAppBridge: MiniAppBridge) {
+    this.bridge = miniAppBridge;
+  }
+
+  getUserName(): Promise<string> {
+    return this.bridge.getUserName();
+  }
+
+  getProfilePhoto(): Promise<string> {
+    return this.bridge.getProfilePhoto();
+  }
+}
+
 /* tslint:disable:no-any */
 export class MiniApp implements MiniAppFeatures, Ad, Platform {
   private bridge: MiniAppBridge = (window as any).MiniAppBridge;
+  private user: UserInfoProvider = new UserInfo(this.bridge);
```

Comment:
I was thinking that this property would be public, so the user data features would be used by calling `MiniApp.user.getUserName()`, and then you don't need to add a third `getUserName` function to this class. But honestly, I'm not sure what's the best interface design to use.

What do you think?

Reply:
I think its a good idea too. I couldn't imagine the number of interfaces added to MiniAppBridge. Sooner the better we break down like this. I think it is good to have it for `Ad` as well before MiniApp starts using them.